File: backend/app/models/prescription.py
from sqlalchemy import Column, Integer, String, ForeignKey, Date
from sqlalchemy.sql import func

from app.database import Base


class Prescription(Base):
    __tablename__ = "prescriptions"

    id = Column(Integer, primary_key=True, index=True)

    patient_id = Column(Integer, ForeignKey("patients.id"))

    doctor_id = Column(Integer, ForeignKey("users.id"))

    medicines = Column(String, nullable=False)

    dosage = Column(String, nullable=False)

    instructions = Column(String, nullable=False)

    prescribed_date = Column(
        Date,
        server_default=func.current_date(),
    )

File: frontend/lib/models/prescription.dart
class Prescription {
  final int? id;
  final int patientId;
  final int doctorId;
  final String medicines;
  final String dosage;
  final String instructions;

  Prescription({
    this.id,
    required this.patientId,
    required this.doctorId,
    required this.medicines,
    required this.dosage,
    required this.instructions,
  });

  factory Prescription.fromJson(Map<String, dynamic> json) {
    return Prescription(
      id: json["id"],
      patientId: json["patient_id"],
      doctorId: json["doctor_id"],
      medicines: json["medicines"],
      dosage: json["dosage"],
      instructions: json["instructions"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "patient_id": patientId,
      "doctor_id": doctorId,
      "medicines": medicines,
      "dosage": dosage,
      "instructions": instructions,
    };
  }
}